### ISSUE TITLE
UNT0033 false positive

### ISSUE THREAD

**Issue by heeen** [microsoft/Microsoft.Unity.Analyzers#262]
#### Bug description
UNT0033 is reported but the case appears to be correct. The quick fix does not change anything.

- Version of analyzers assembly: 79a79a9 built myself
- Analyzer rule: UNT0033
- Error (exception message, type, and callstack where applicable): `warning UNT0033: The Unity message 'OnPostprocessAllAssets' has an incorrect case`

#### To Reproduce
Steps or code to reproduce the behavior:
```
class DeletionDetector : AssetPostprocessor
{
    static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
    {
    }
}
```
you can also use the code from the unity docs: https://docs.unity3d.com/ScriptReference/AssetPostprocessor.OnPostprocessAllAssets.html

#### Expected behavior
This should not cause the diagnostic to be generated.

#### Additional context
testing on linux/vscode, the diagnostic appears in both unity editor and vscode/omnisharp.

**Comment by heeen** [comment]
So it looks like all unit tests fail on linux because it fails to locate a unity installation in the first place.

**Comment by heeen** [comment]
I fixed the detection of linux installations and the bug persists because the list of all messages contains duplicates:

> all messages:
>  OnAssignMaterialModel, OnPostprocessAnimation, OnPostprocessAssetbundleNameChanged, OnPostprocessAudio, OnPostprocessCubemap, OnPostprocessGameObjectWithAnimatedUserProperties, OnPostprocessGameObjectWithUserProperties, OnPostprocessMaterial, OnPostprocessMeshHierarchy, OnPostprocessModel, OnPostprocessSpeedTree, OnPostProcessSprites, OnPostprocessTexture, OnPostprocessPrefab, OnPostprocessTexture2DArray, OnPostprocessTexture3D, OnPreprocessAnimation, OnPreprocessAsset, OnPreprocessAudio, OnPreprocessMaterialDescription, OnPreprocessModel, OnPreprocessSpeedTree, OnPreprocessTexture, OnPreprocessCameraDescription, OnPreprocessLightDescription, **OnPostprocessAllAssets, OnPostprocessAllAssets,** OnPreGeneratingCSProjectFiles, OnGeneratedSlnSolution, OnGeneratedCSProject XXX
> XXX not implemented messages:
> 
> OnAssignMaterialModel, OnPostprocessAnimation, OnPostprocessAssetbundleNameChanged, OnPostprocessAudio, OnPostprocessCubemap, OnPostprocessGameObjectWithAnimatedUserProperties, OnPostprocessGameObjectWithUserProperties, OnPostprocessMaterial, OnPostprocessMeshHierarchy, OnPostprocessModel, OnPostprocessSpeedTree, OnPostProcessSprites, OnPostprocessTexture, OnPostprocessPrefab, OnPostprocessTexture2DArray, OnPostprocessTexture3D, OnPreprocessAnimation, OnPreprocessAsset, OnPreprocessAudio, OnPreprocessMaterialDescription, OnPreprocessModel, OnPreprocessSpeedTree, OnPreprocessTexture, OnPreprocessCameraDescription, OnPreprocessLightDescription, OnPostprocessAllAssets, OnPreGeneratingCSProjectFiles, OnGeneratedSlnSolution, OnGeneratedCSProject XXX
>

**Comment by heeen** [comment]
the diagnostic does not handle overloads correctly:
```
Void OnPostprocessAllAssets(System.String[], System.String[], System.String[], System.String[])
Void OnPostprocessAllAssets(System.String[], System.String[], System.String[], System.String[], Boolean)
```
if I implement either of those, the other will be in the not implemented list matching the name comparison.

**Comment by sailro** [comment]
Thank you for the report and for the fix. Your fix is working, but here given we are only working on the name (we have another analyzer for signatures), we can fix it without matching parameters.

See #268
